### PR TITLE
Write and symlink kubeconfig on change of target

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
With this PR it is possible to change the target cluster of `kubectl` directly with the `gardenctl target ...` command without having to adjust the `KUBECONFIG` environment variable again and again. Whether the current target cluster of gardenctl and kubectl should be synchronized or not can be set using the configrations property link-kubeconfig or via environment variable `GCTL_LINK_KUBECONFIG`.

#### Behavior without symlink | linkKubeconfig: false
The `target` command only changes the current target cluster of gardenctl, but does not download the corresponding kubeconfig file. This happens only when the `kubectl-env` command is executed. If the user evals the generated script the `KUBECONFIG` environment variable points the downloaded kubeconfig file for the targeted cluster.
Link-Kubeconfig-2
The generated script looks something like this:
```bash
export KUBECONFIG=$TMPDIR/garden/<gctl-session-id>/kubeconfig.<hash>.yaml
```

#### Behavior with symlink | linkKubeconfig: true
The `target` command changes the current target cluster of gardenctl, downloads the corresponding kubeconfig file and creates a stable symlink to the current kubeconfig file in the session directory. If the target is changed the symlink points to the new target but the `KUBECONFIG` environment variable does not need to be changed. The `kubectl-env` command now does nothing except create the script to set the `KUBECONFIG` environment variable. This works even if nothing is targeted and could be done in the shell profile.
Link-Kubeconfig-1
The generated script only needs to be evaluated once per shell session and looks something like this:
```bash
export KUBECONFIG=$TMPDIR/garden/<gctl-session-id>/kubeconfig.yaml
```
#### Test
Test are not yet changed !!!

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Comment by RaphaelVogel** [comment]
nice, waiting to get this feature 👍

File: pkg/cmd/env/options.go [hunk 1]
```diff
@@ -140,14 +148,20 @@ func (o *options) runKubernetes(ctx context.Context, manager target.Manager) err
...
-		if err != nil {
-			return err
-		}
+		var filename string
 
-		filename, err := manager.WriteClientConfig(config)
-		if err != nil {
-			return err
+		if o.Symlink {
+			filename = path.Join(o.SessionDir, "kubeconfig.yaml")
```
Comment (holgerkoser) [inline pkg/cmd/env/options.go:154]: Add error message if target exists but no symlink

File: pkg/config/config.go [hunk 2]
```diff
@@ -56,6 +59,14 @@ func LoadFromFile(filename string) (*Config, error) {
 
 	config := &Config{Filename: filename}
 
+	// We don't want a dependency to root command here
+	str, ok := os.LookupEnv("GCTL_LINK_KUBECONFIG")
+	if ok {
+		if val, err := strconv.ParseBool(str); err == nil {
```
Comment (petersutter) [inline pkg/config/config.go:65]: shouldn't we let the user know if `GCTL_LINK_KUBECONFIG` could not be parsed to bool indicating a wrong value or just silently ignore?

File: pkg/config/config.go [hunk 3]
```diff
@@ -21,6 +22,8 @@ import (
 type Config struct {
 	// Filename is the name of the gardenctl configuration file
 	Filename string `yaml:"-"`
+	// LinkKubeconfig defines if kubeconfig is symlinked with the target
+	LinkKubeconfig *bool `yaml:"link-kubeconfig,omitempty"`
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply pkg/config/config.go:26]: Do you prefer camelCase? In kubeconfig we have `certificate-authority-data` or `current-context` but I know in gardener we also have camelCase.

[the remark this answers is not in the document]
Comment (holgerkoser) [reply pkg/config/config.go:26]: I have change it to `linkKubeconfig`

File: pkg/cmd/env/options.go (hunk 1 above)
Comment (holgerkoser) [reply pkg/cmd/env/options.go:154]: ✔️

**Review by petersutter — APPROVED** [review]
/lgtm

**Pull request merged**